charts/agent-control/templates/rbac.yaml

```diff
@@ -46,7 +47,63 @@ roleRef:
   kind: Role
   name: {{ include "newrelic.common.naming.truncateToDNSWithSuffix" (dict "name" (include "newrelic.common.naming.fullname" .) "suffix" "install-job-resources") }}
 subjects:
-- kind: ServiceAccount
-  name: {{ include "newrelic.common.naming.truncateToDNSWithSuffix" (dict "name" .Release.Name "suffix" "install-job") }}
+  - kind: ServiceAccount
+    name: {{ include "newrelic.common.naming.truncateToDNSWithSuffix" (dict "name" .Release.Name "suffix" "install-job") }}
+    namespace: {{ .Release.Namespace }}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  labels:
+    {{- include "newrelic.common.labels" . | nindent 4 }}
+  name: {{ include "newrelic.common.naming.truncateToDNSWithSuffix" (dict "name" (include "newrelic.common.naming.fullname" .) "suffix" "uninstall-job-resources") }}
   namespace: {{ .Release.Namespace }}
+rules:
+  - apiGroups: [ "" ]
+    resources: [ "namespaces" ]
+    verbs:
+      - get
+  - apiGroups: [ "apps" ]
+    resources: [ "deployments", "daemonsets", "statefulsets" ]
+    verbs:
+      - list
+      - watch
+  - apiGroups: [ "source.toolkit.fluxcd.io", "helm.toolkit.fluxcd.io" ]
+    resources: [ "helmrepositories", "helmreleases" ]
+    verbs:
+      - list
+      - watch
+      - delete
+      - deletecollection
+  - apiGroups: [ "newrelic.com" ]
+    resources: [ "*" ]
+    verbs:
+      - delete
+      - deletecollection
+      - get
+      - list
+      - watch
+  - apiGroups: [ "" ]
+    resources: [ "configmaps","secrets" ]
+    verbs:
+      - delete
+      - list
+      - watch
+      - deletecollection
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
```

Comment:
I would follow the same approach as above, in the installation job we have the hooks, no?

Reply:
Yes, but these are used for the uninstallation job. They are automatically removed when we delete the helm release. So... Do we need the hooks at all?

Reply:
Ah, I should use the same hooks as in the service account, right?